lua/lualine/components/signify.lua

```diff
@@ -1,33 +1,72 @@
--- Copyright (c) 2020-2021 hoob3rt
+-- Copyright (c) 2020-2021 shadmansaleh
 -- MIT license, see LICENSE for more details.
 
-local function signify()
-   if vim.fn.exists('*sy#repo#get_stats') == 0 then return '' end
-   local added, modified, removed = unpack(vim.fn['sy#repo#get_stats']())
-   if added == -1 then return '' end
-   local symbols = {
-     '+',
-     '-',
-     '~',
-   }
-   local result = {}
-   local data = {
-    added,
-    removed,
-    modified,
-   }
-   for range=1,3 do
-     if data[range] ~= nil and data[range] > 0
-       then table.insert(result,symbols[range]..''..data[range]..' ')
-     end
-   end
-
-   if result[1] ~= nil then
-       return table.concat(result, '')
-   else
-       return ''
-   end
-end
+local utils = require'lualine.utils.utils'
+
+local function signify(args)
+  local colored = true
+  if args.colored ~= nil then colored = args.colored end
+
+  -- apply defaults
+  local color_modified = "#f0e130"
```

Comment:
Wouldn't it be better to have
```
if color_modified ==nil then color_modified = '%#DiffAdded`
```
etc. Most themes already define diff colors. I think having hardcoded one is bad.

Reply:
Good idea . Using colorscemes highlight didn't come to my mind

Reply:
Ok I've looked in it and I donn't like it.
First those highlight groups background are based on colorschemes background so they donn't blend with statuslines theme (looks oweful actualy). But when we create highlight groups it is created based on satuslines
theme so it blends with the background .
Secondly if the colorscheme doesn't support cterm colors our support for cterm colors goes out the window too.
Also I think it added unnesery complexcity as if we default to those hl_groups we might have to create some groups and not really on colorcheme for others . 

Also since colors are fully customizable haveing defaults hardcoded isn't that big of a deal in my opinion.

Reply:
thanks for pointing that out. I've pushed a solution to this problem. I think it works well now

Reply:
I think you're extracting fg colors . checkout #94 highlight.lua in get transitional_highlights function I thunk thats a better approch than trying to parse output.

Reply:
If you know how to improve it then go ahead. I knew there was a vim.api command to get highlight but forgot to do it here :neutral_face:

Reply:
I'll do it . I was just passing the idea :)